src/js/actions/documents.js

```diff
@@ -856,12 +860,119 @@ define(function (require, exports) {
+        var appStore = this.flux.store("application"),
+            docStore = this.flux.store("document");
+
+        /**
+         * Make list of currently open documents to be used as search options
+         * 
+         * @return {Immutable.List.<object>}
+        */
+        var currDocSearchOptions = function () {
+            var document = appStore.getCurrentDocument(),
+                openDocs = appStore.getOpenDocumentIDs().filterNot(function (doc) {
+                                return doc === document.id;
+                            }),
+                docMap = openDocs.map(function (doc) {
+                    return {
+                        id: "curr-doc_" + doc.toString(),
+                        title: docStore.getDocument(doc).name,
+                        type: "item",
+                        category: [strings.SEARCH.CATEGORIES.DOCUMENT, strings.SEARCH.CATEGORIES.CURRENT]
+                    };
+                }),
+                docLabel = {
+                    id: "curr-doc_header",
+                    title: strings.SEARCH.HEADERS.CURRENT_DOC,
+                    type: "header"
+                },
+
+                docOptions = docMap.unshift(docLabel);
+
+            return docOptions;
+        };
+
+
+        /**
+         * Make list of recent documents to be used as search options
+         * 
+         * @return {Immutable.List.<object>}
+        */
+        var recentDocSearchOptions = function () {
+            var recentFiles = appStore.getRecentFiles(),
+                recentDocMap = recentFiles.map(function (doc, index) {
+                    return {
+                        id: "recent-doc_" + index.toString(),
+                        title: pathUtil.getShortestUniquePaths(Immutable.List.of(doc)).toJS()[0],
+                        type: "item",
+                        info: doc,
+                        displayInfo: doc,
+                        category: [strings.SEARCH.CATEGORIES.DOCUMENT, strings.SEARCH.CATEGORIES.RECENT]
+                    };
+                });
+            
+            // Get shortest unique file path
+            var paths = collection.pluck(recentDocMap, "info"),
+                shortenedPaths = pathUtil.getShortestUniquePaths(paths).toJS();
+
+            recentDocMap.forEach(function (docItem, index) {
+                docItem.displayInfo = shortenedPaths[index];
+            });
+
+            var recentDocLabel = {
+                id: "recent-doc_header",
+                title: strings.SEARCH.HEADERS.RECENT_DOC,
+                type: "header"
+            };
+
+            return recentDocMap.unshift(recentDocLabel);
+        };
+
+        var getDocumentSearchOptions = function () {
+            return recentDocSearchOptions().concat(currDocSearchOptions());
+        };
+
+        var confirmSearch = function (itemID) {
+            // ID has type as first word, followed by the layer/document ID
+            var idArray = itemID.split("_"),
```

Comment:
I really like this logic of returning false if it's not being handled. But, I think it would be better if the id=>handleExecute mapping is taken care of by the store, so it can just pass by the actual 'idInt' in here where all you'd need to do is call the handling function.